route/middleware.go

```diff
@@ -58,6 +59,26 @@ func (r *Router) apiKeyChecker(next http.Handler) http.Handler {
 	})
 }
 
+func (r *Router) apiKeyCheckerOTLP(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		apiKey := req.Header.Get(types.APIKeyHeader)
+		if apiKey == "" {
+			apiKey = req.Header.Get(types.APIKeyHeaderShort)
+		}
+		if apiKey == "" {
+			err := errors.New("no " + types.APIKeyHeader + " header found from within authing middleware")
+			r.handleOTLPFailureResponse(w, req, husky.OTLPError{Message: err.Error(), HTTPStatusCode: http.StatusUnauthorized})
+			return
+		}
+		if r.Config.IsAPIKeyValid(apiKey) {
+			next.ServeHTTP(w, req)
+			return
+		}
+		err := fmt.Errorf("api key %s not found in list of authorized keys", apiKey)
```

Comment:
Same here, I don't think we need to add the API key into the response message as they know what they set. This could be a const too.

Reply:
Removed entirely